pulsar-broker/src/test/java/org/apache/pulsar/common/naming/TopicCountEquallyDivideBundleSplitAlgorithmTest.java

```diff
@@ -0,0 +1,93 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.naming;
+
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import com.google.common.hash.Hashing;
+import org.apache.pulsar.broker.namespace.NamespaceService;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.testng.Assert;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+
+public class TopicCountEquallyDivideBundleSplitAlgorithmTest {
+
+    @Test
+    public void testWrongArg() {
+        TopicCountEquallyDivideBundleSplitAlgorithm algorithm = new TopicCountEquallyDivideBundleSplitAlgorithm();
+        Assert.assertThrows(NullPointerException.class, () -> algorithm.getSplitBoundary(null, null));
+    }
+
+    @Test
+    public void testTopicsSizeLessThan1() {
+        TopicCountEquallyDivideBundleSplitAlgorithm algorithm = new TopicCountEquallyDivideBundleSplitAlgorithm();
+        NamespaceService mockNamespaceService = Mockito.mock(NamespaceService.class);
+        NamespaceBundle mockNamespaceBundle = Mockito.mock(NamespaceBundle.class);
+        Mockito.doReturn(CompletableFuture.completedFuture(Lists.newArrayList("a")))
+                .when(mockNamespaceService).getOwnedTopicListForNamespaceBundle(mockNamespaceBundle);
+        Assert.assertNull(algorithm.getSplitBoundary(mockNamespaceService, mockNamespaceBundle).join());
+    }
+    @Test
+    public void testAlgorithmReturnCorrectResult() {
+        TopicCountEquallyDivideBundleSplitAlgorithm algorithm = new TopicCountEquallyDivideBundleSplitAlgorithm();
+        List<String> mockTopics = Lists.newArrayList("a", "b", "c");
+        long correctResult = getCorrectResult(mockTopics);
+        NamespaceService mockNamespaceService1 = Mockito.mock(NamespaceService.class);
+        NamespaceBundle mockNamespaceBundle1 = Mockito.mock(NamespaceBundle.class);
+        Mockito.doReturn(CompletableFuture.completedFuture(mockTopics))
+                .when(mockNamespaceService1).getOwnedTopicListForNamespaceBundle(mockNamespaceBundle1);
+        NamespaceBundleFactory mockNamespaceBundleFactory = Mockito.mock(NamespaceBundleFactory.class);
+        mockTopics.forEach((topic)->{
+            Mockito.doReturn(mockNamespaceBundleFactory)
+                    .when(mockNamespaceBundle1).getNamespaceBundleFactory();
+            long hashValue = Hashing.crc32().hashString(topic, Charsets.UTF_8).padToLong();
+            Mockito.doReturn(hashValue)
+                    .when(mockNamespaceBundleFactory).getLongHashCode(topic);
+        });
+        Assert.assertEquals((long)algorithm.getSplitBoundary(mockNamespaceService1, mockNamespaceBundle1).join(),correctResult);
+    }
+
+    private long getCorrectResult(List<String> mockTopics){
```

Comment:
"getCorrectResult" is a very vague name for a method.
It seems that a better approach would be to inline this method.

Reply:
OK, thank you, I will fix it.  : )